**Python Assignment 2**

**Data Structures - List, Dictionary, Set
& Conditional Statements**

1.List Creation:

 a. Create a list named age_list with five integer elements. For eg., [24, 25, 26, 27, 28]

In [ ]:
age_list = [24,25,26,27,28]
print(age_list)

[24, 25, 26, 27, 28]


b. Create a list named name_list with five string elements.

In [ ]:
name_list = ["Layana", "Akash", "Gloria", "Archa", "Sudheesh"]
print(name_list)

['Layana', 'Akash', 'Gloria', 'Archa', 'Sudheesh']


* In a and b section demonstrates list creation using square brackets to store multiple values in one variable. age_list holds five integers and name_list holds five strings, each printed to verify the contents.

2.List Operations / Modifications:

a. Append the string "Yazhini" to name_list.

In [ ]:
name_list.append("Yazhini")
print(name_list)

['Layana', 'Akash', 'Gloria', 'Archa', 'Sudheesh', 'Yazhini']


In [ ]:
# b. Insert the element 30 at index 2 in age_list.

age_list.insert(2, 30)
print(age_list)

[24, 25, 30, 26, 27, 28]


In [ ]:
# c. Remove the string "Yazhini" from name_list.

name_list.remove("Yazhini")
print(name_list)

['Layana', 'Akash', 'Gloria', 'Archa', 'Sudheesh']


In [ ]:
# d. Pop the last element from age_list.

age_list.pop()
print(age_list)


[24, 25, 30, 26, 27]


In [ ]:
# e. Extend the age_list with additional ages [29, 30, 26].

age_list.extend([29,30,26])
print(age_list)

[24, 25, 30, 26, 27, 29, 30, 26]


In [ ]:
# f. Sort age_list in descending order.

age_list.sort(reverse=True)
print(age_list)

[30, 30, 29, 27, 26, 26, 25, 24]


In [ ]:
# g. Find Max age, Min age and sum of all ages from age_list.

print("Max age:", max(age_list))
print("Min age:", min(age_list))
print("Sum of ages:", sum(age_list))

Max age: 30
Min age: 24
Sum of ages: 217


a) "Yazhini" added to the end of list, name_list.append("Yazhini")
* Output: ['Layana', 'Akash', 'Gloria', 'Archa', 'Sudheesh', 'Yazhini']

b) Inserted 30 at position 2 and shifts others right, age_list.insert(2,30)
* Output: [24, 25, 30, 26, 27, 28]

c) To remove "Yazhini" from string, name_list.remove("Yazhini")
* Output: ['Layana', 'Akash', 'Gloria', 'Archa', 'Sudheesh']

d) To pop the last element from age list, age_list.pop()
* Output: [24, 25, 30, 26, 27], removes and returns the last element

e) To extend the age_list, age_list.extend([29,30,26])
* Output: [24, 25, 30, 26, 27, 29, 30, 26] Adds multiple elements to end

f) To sort age_list in descending order, age_list.sort(reverse=True)
* Output: [30, 30, 29, 27, 26, 26, 25, 24] Sorts high to low

g) To find max, min and sum of all ages
* max(age_list): Returns the largest element in the list
* Output= Max age:30
* min(age_list): Returns the smallest element in the list
* Output= Min age: 24
* sum(age_list): Returns the total of all numeric elements in the list
* Output= Sum of ages: 217

3.Accessing List Elements:


In [ ]:
# a. Print the first element of name_list.

print(name_list[0])

Layana


In [ ]:
# b. Print the last element of name_list.

print(name_list[-1])

Sudheesh


In [ ]:
# c. Print the elements from index 2 to index 4 in name_list.

print(name_list[2:5])

['Gloria', 'Archa', 'Sudheesh']


In [ ]:
# d. Print the elements of name_list in reverse order.

print(name_list[::-1])

['Sudheesh', 'Archa', 'Gloria', 'Akash', 'Layana']


a) [0] prints the first element "Layana"

b) [-1] prints the last element "Sudheesh"

c) [2:5] prints index 2 to 4 as a slice
* Output: ['Gloria', 'Archa', 'Sudheesh']

d) [::-1] prints the list in reverse order.
* Output: ['Sudheesh', 'Archa', 'Gloria', 'Akash', 'Layana']

**Dictionary (Creation, Modification and Access):**


a. Create a dictionary named student_marks that maps the names of five
students to their marks (use scale of from 0 to 100).

In [ ]:
student_marks = {"Layana": 85, "Akash": 90, "Gloria": 78, "Archa": 95, "Sudheesh": 75}
print(student_marks)

{'Layana': 85, 'Akash': 90, 'Gloria': 78, 'Archa': 95, 'Sudheesh': 75}


In [ ]:
# b. Access and print the mark of a specific student, of your choice.

print(student_marks["Akash"])

90


In [ ]:
# c. Add a new student "Janani" with a mark of 80 to the student_marks dictionary.

student_marks["Janani"] = 80
print(student_marks)

{'Layana': 85, 'Akash': 90, 'Gloria': 78, 'Archa': 95, 'Sudheesh': 75, 'Janani': 80}


In [ ]:
# d. Update the mark of any one older student to 82.

student_marks["Layana"]= 82
print(student_marks)

{'Layana': 82, 'Akash': 90, 'Gloria': 78, 'Archa': 95, 'Sudheesh': 75, 'Janani': 80}


In [ ]:
# e. Use the keys(), values(), and items() methods to print all keys, values, and key-value pairs in the student_marks dictionary.

print(student_marks.keys())
print(student_marks.values())
print(student_marks.items())

dict_keys(['Layana', 'Akash', 'Gloria', 'Archa', 'Sudheesh', 'Janani'])
dict_values([82, 90, 78, 95, 75, 80])
dict_items([('Layana', 82), ('Akash', 90), ('Gloria', 78), ('Archa', 95), ('Sudheesh', 75), ('Janani', 80)])


a) Created student_marks = {"Layana": 85, "Akash": 90, "Gloria": 78, "Archa": 95, "Sudheesh": 75}, dictionary mapping five student names to marks

b) print(student_marks["Akash"]) Accesses a specific student's mark using the key "Akash" and output: 90

c) student_marks["Janani"] = 80, Adds a new key value pair "Janani":80 to the dictionary

d) student_marks["Layana"]= 82, Updates Layana's mark to 82 by reassigning a new value to the existing key.

e) print(student_marks.keys()), print(student_marks.values()), print(student_marks.items()), Uses keys(),values() and items() to print all student names, all marks and all name-mark pairs.

**Sets (Operations):**

a. Create a set called my_set with following values:    
['a','e','i','o','u','a','a','i']
Analyse the output and provide explanation for the same.

In [ ]:
my_set = {'a','e','i','o','u','a','a','i'}
print(my_set)

{'e', 'i', 'u', 'o', 'a'}


a) Answer
* Here duplicates are automatically removed.
* Output: {'e', 'i', 'u', 'o', 'a'}, because sets store only unique elements.

In [ ]:
# b. Attempt to change the value of my_set[4] = 's'. If code throws an error, provide an explanation.

my_set[4]='s'
print(my_set)

TypeError: 'set' object does not support item assignment

b) Answer:
* This error happens because sets are unordered so the indexing will not work in case of sets, so we can't access or change elements by position.

In [ ]:
# c. Create two sets:
# set1 with values: {1, 3, 5, 7, 9}
# set2 with values: {2, 3, 5, 8, 10}

set1 = {1,3,5,7,9}
set2 = {2,3,5,8,10}
print(set1)
print(set2)

{1, 3, 5, 7, 9}
{2, 3, 5, 8, 10}


In [ ]:
# d. Compute and print the union and intersection of set1 and set2.

print(set1.union(set2))  #Union
print(set1.intersection(set2)) #intersection

{1, 2, 3, 5, 7, 8, 9, 10}
{3, 5}


c) Created two sets set1 and set2 using curly braces to store unique unordered values.

d) print(set1.union(set2)), Computes the union with union() for all distinct elements
* Output: {1, 2, 3, 5, 7, 8, 9, 10}
* print(set1.intersection(set2)), intersection() takes common elements
* Output: {3,5}

**Operators & Conditional Statements :
(IF, ELIF, ELSE)**

Performance Category Program:
1. Prompt user for Input. Score range should be from 0 to 10 (both inclusive).
2. Find the performance category based on the input score using following criteria:
a. Above Average:  Score greater than 7
b. Average: Score between 4 and 7(both inclusive)
c. Below Average: Score lesser than 4
3. Output: Print the Performance category
4. Additional Step: You can give a prompt of your choice to each category.  
For eg: If score below average “Need to Improve your performance, consistent
practice will lead to better results”.  

In [ ]:
score = int(input("Enter your score(0 to 10): "))
if score >7:
  print("Above Average: Excellent work! Keep it up.")
elif score >=4:
  print("Average: Good effort! Keep practicing, there's little more steps for make excellent.")
else:
  print("Below Average: Need to improve your performance, consistent practice will lead to better results.")

Enter your score(0 to 10): 8
Above Average: Excellent work! Keep it up.


This program takes a score from 0 to 10 as input and categorizes performance using if-elif-else. Score >7 prints Above Average, 4 to 7 prints Average and below 4 prints Below Average with a motivational message.